src/Utilities/Compiler/Options/SymbolNamesOption.cs

```diff
@@ -85,13 +103,16 @@ public static SymbolNamesOption Create(ImmutableArray<string> symbolNames, Compi
                 return Empty;
             }
 
-            return new SymbolNamesOption(namesBuilder.ToImmutableAndFree(), symbolsBuilder.ToImmutableAndFree());
+            return new SymbolNamesOption(namesBuilder.ToImmutableDictionaryAndFree(), symbolsBuilder.ToImmutableDictionaryAndFree());
         }
 
         public bool IsEmpty => ReferenceEquals(this, Empty);
 
         public bool Contains(ISymbol symbol)
-            => _symbols.Contains(symbol) || _names.Contains(symbol.Name);
+            => _symbols.ContainsKey(symbol) || _names.ContainsKey(symbol.Name);
+
+        public bool TryGetSuffix(ISymbol symbol, [NotNullWhen(true)] out string? suffix) =>
```

Comment:
```suggestion
        public bool TryGetValue(ISymbol symbol, [NotNullWhen(true)] out string? value) =>
```

Reply:
Please also add a doc comment that this method will fetch the string value associated with the given symbol in the option specification.